app/services/fare_calculator_service.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class FareCalculatorService
+  attr_reader :journey
+
+  def initialize(journey)
+    @journey = journey
+  end
+
+  def calculate
+    return Fare.bus_journey.first.amount if journey.bus_journey?
+
+    # journey.preload(start_station: :zones, end_station: :zones)
+    calculate_train_fare
+  end
+
+  private
+
+  def calculate_train_fare
+    case journey_zones_count
+    when 1
+      single_zone_fare
+    when 2
+      two_zone_fare
+    when 3
+      three_zone_fare
+    end
```

Comment:
Think in the case `journey_zones_count` is a different value, other than 1,2 or 3 you'll have a runtime error. Is a good practice define a default value just in case

```suggestion
    case journey_zones_count
    when 1
      single_zone_fare
    when 2
      two_zone_fare
    when 3
      three_zone_fare
    else
      default_zone_fare
    end
```